### ISSUE TITLE
Libraries FullMath.sol and FixedPoint<96,128> don't work for Solidity >=0.8.0

### ISSUE THREAD

**Issue by rrodriguezreche** [Uniswap/uniswap-v3-core#489]
I suggest limiting these with `pragma solidity >= 0.5.0 <0.8.0`.

**Comment by TbLtzk** [comment]
Agree! We are using your library for our project and are about to migrate to v0.8.9.

Do you consider a second branch of your library (and npm publication, e.g. with major version increased, i.e. 2.x.x)?
If so, we would be glad to contribute a `>= v0.8.0` version of `FullMath.sol`

Thanks for this great piece of code, btw. If you want to learn more about our project, head over to https://q.org or https://gitlab.com/q-dev

**Comment by gzliudan** [comment]
I encounter this error also when I use solidity 0.8.x :
```text
TypeError: Unary operator - cannot be applied to type uint256
  --> @uniswap/v3-core/contracts/libraries/FullMath.sol:64:24:
   |
64 |         uint256 twos = -denominator & denominator;
```

**Comment by h0tw4t3r** [comment]
TickMath as well
